src/sidebar/store/modules/real-time-updates.js

```diff
@@ -116,7 +119,8 @@ function receiveRealTimeUpdates({
       // even if the annotation is from the current group, it might be for a
       // new annotation (saved in pendingUpdates and removed above), that has
       // not yet been loaded.
-      if (annotations.selectors.annotationExists(getState(), id)) {
+      const annotationsState = getState().annotations;
```

Comment:
Does this need to be done on each iteration here? Can it be grouped with `groupState` and `routeState` above? Or is there state that is changing as we iterate here that I'm not grokking. If it does need to stay here, can you add a comment as to why?

Reply:
This _could_ be hoisted out of the loop, although `getState()` is a super cheap call. In fact the JS VM might even inline it. It doesn't matter that much, but moving it to the top of the loop could create a small refactoring hazard if someone were to insert a `dispatch` call in the function which modified the state. Having it here ensures that we're definitely using the latest state.